numpy/lib/arraysetops.py

```diff
@@ -217,10 +254,35 @@ def unique(ar, return_index=False, return_inverse=False,
     >>> u[indices]
     array([1, 2, 6, 4, 2, 3, 2])
 
+    Obtain unique occurrences with the least amount of overhead
+    (sort in-place, instead on a temporary copy):
+
+    >>> a = np.array([1, 2, 3, 1, 2, 3, 1, 2, 3])
+    >>> a.sort()
+    >>> u = np.unique(a, assume_sorted=True)
+    >>> u
+    array([1,2,3])
+
+    >>> a = np.array([1, 2, 3, 1, 2, 3, 1, 2, 3])
+    >>> u = np.unique(a, sort_inplace=True)
+    >>> u
+    array([1,2,3])
+    >>> a
+    array([1,1,1,2,2,2,3,3,3])
+
     """
+    orig_ar = ar
     ar = np.asanyarray(ar)
+
+    indirect_sort = return_index or return_inverse
+    if indirect_sort:
+        # We won't perform an in-place sort even if requested
+        sort_inplace = False
```

Comment:
Why not do one, if it was requested explictly?

Reply:
`return_index` and `return_inverse` both require an argsort, not an inplace sort. Getting the original array in sorted order while still computing the indexes or the inverse is really cumbersome and inefficient, and not all that useful. So they weren't implemented.

Basically, sort_inplace is more a permission to do the sort in place than a request. It means: it's ok to modify the original sort order.